crates/nu-cli/tests/completions/mod.rs

```diff
@@ -363,6 +400,146 @@ fn file_completions() {
     match_suggestions(&expected_paths, &suggestions);
 }
 
+#[derive(Clone)]
+pub struct CustomCommandWithAnyArgs;
+
+impl Command for CustomCommandWithAnyArgs {
+    fn name(&self) -> &str {
+        "list"
+    }
+
+    fn description(&self) -> &str {
+        "Mock def command."
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build("list").rest("args", SyntaxShape::Any, "rest of the arguments")
+    }
+
+    fn run(
+        &self,
+        _engine_state: &EngineState,
+        _stack: &mut Stack,
+        _call: &Call,
+        _input: PipelineData,
+    ) -> Result<PipelineData, ShellError> {
+        todo!()
+    }
+}
+
+#[test]
+fn custom_command_completions() {
+    // Create a new engine
+    let (dir, dir_str, mut engine, stack) = new_engine();
+    let delta = {
+        let mut working_set = StateWorkingSet::new(&engine);
+        working_set.add_decl(Box::new(CustomCommandWithAnyArgs));
+        working_set.render()
+    };
+    engine.merge_delta(delta).expect("Failed to merge delta");
+
+    // Instantiate a new completer
+    let mut completer = NuCompleter::new(Arc::new(engine), Arc::new(stack));
+
+    // Test completions for the current folder
+    let target_dir = format!("list {dir_str}{MAIN_SEPARATOR}");
+    let suggestions = completer.complete(&target_dir, target_dir.len());
+
+    // Create the expected values
+    let expected_paths: Vec<String> = vec![
+        folder(dir.join("another")),
+        file(dir.join("custom_completion.nu")),
+        folder(dir.join("directory_completion")),
+        file(dir.join("nushell")),
+        folder(dir.join("test_a")),
+        folder(dir.join("test_b")),
+        file(dir.join(".hidden_file")),
+        folder(dir.join(".hidden_folder")),
+    ];
+
+    #[cfg(windows)]
+    {
+        let separator = '/';
+        let target_dir = format!("list {dir_str}{separator}");
+        let slash_suggestions = completer.complete(&target_dir, target_dir.len());
+
+        let expected_slash_paths: Vec<String> = expected_paths
+            .iter()
+            .map(|s| s.replace('\\', "/"))
+            .collect();
+
+        match_suggestions(&expected_slash_paths, &slash_suggestions);
+    }
+
+    // Match the results
+    match_suggestions(&expected_paths, &suggestions);
```

Comment:
Is this first check necessary? Completions for the first argument of a custom command are already tested in other places, right?

Reply:
Similar answer as: https://github.com/nushell/nushell/pull/14481#discussion_r1864429945

Reply:
Sure, let's leave this check in. Although, like I said in https://github.com/nushell/nushell/pull/14481#discussion_r1865063319, I don't think the `#[cfg(windows)] {...}` bit is necessary. As long as we know file completions are triggered for custom commands (on the first argument and on later arguments) and that the Windows `/` behavior works for internal commands, I think we can trust that the `/` behavior will work for custom commands too.

Reply:
I read this as I could resolve this thread now.